plugins/catalog-backend/src/database/Database.ts

```diff
@@ -14,33 +14,43 @@
  * limitations under the License.
  */
 
-import { InputError, NotFoundError } from '@backstage/backend-common';
+import {
+  ConflictError,
+  InputError,
+  NotFoundError,
+} from '@backstage/backend-common';
 import Knex from 'knex';
+import lodash from 'lodash';
 import { v4 as uuidv4 } from 'uuid';
-import { DescriptorEnvelope } from '../ingestion';
+import { Logger } from 'winston';
+import { DescriptorEnvelope, EntityMeta } from '../ingestion';
+import { buildEntitySearch } from './search';
 import {
   AddDatabaseLocation,
   DatabaseLocationUpdateLogEvent,
   DatabaseLocationUpdateLogStatus,
   DbEntitiesRow,
+  DbEntitiesSearchRow,
   DbEntityRequest,
   DbEntityResponse,
   DbLocationsRow,
 } from './types';
 
-function serializeMetadata(
-  metadata: DescriptorEnvelope['metadata'],
-): DbEntitiesRow['metadata'] {
-  if (!metadata) {
-    return null;
-  }
-
-  const output = { ...metadata };
+function getStrippedMetadata(metadata: EntityMeta): EntityMeta {
```

Comment:
Yeah there are a bunch of free floating functions here. I would like to move them to a form of utils file later, but not as part of this PR. Wanna get it in to unblock people :)